File: solutions/csharp/calculator-conundrum/1/CalculatorConundrum.cs
public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string? operation)
    {
        return operation switch 
        {
            "+" => $"{operand1} {operation} {operand2} = {operand1 + operand2}",
            "*" => $"{operand1} {operation} {operand2} = {operand1 * operand2}",
            "/" => operand2 == 0 
                ? "Division by zero is not allowed." 
                : $"{operand1} {operation} {operand2} = {operand1 / operand2}", 
            "" => throw new ArgumentException(),
            null => throw new ArgumentNullException(),
            _ => throw new ArgumentOutOfRangeException()
        };
    }
}

File: solutions/java/calculator-conundrum/1/src/main/java/CalculatorConundrum.java
class CalculatorConundrum {
    public String calculate(int operand1, int operand2, String operation) {
        int result;
        
        if (operation == null) throw new IllegalArgumentException("Operation cannot be null");
        if (operation.isEmpty()) throw new IllegalArgumentException("Operation cannot be empty");

        switch (operation) {
            case "+":
                result = operand1 + operand2;
                break;
            case "*":
                result = operand1 * operand2;
                break;
            case "/":
                try {
                    result = operand1 / operand2;
                } catch (ArithmeticException e) {
                    throw new IllegalOperationException("Division by zero is not allowed", e);
                }
                break;
            default:
                throw new IllegalOperationException("Operation '%s' does not exist".formatted(operation));
        }
        
        return "%d %s %d = %d".formatted(operand1, operation, operand2, result);
    }
}